backend/app/store/search/internal/elastic.go

```diff
@@ -0,0 +1,364 @@
+	resp, err := e.client.Search(
+		e.client.Search.WithIndex(req.SiteID),
+		e.client.Search.WithBody(e.buildQuery(req)),
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if err := checkElasticResponseErr(resp); err != nil {
+		return nil, err
+	}
+
+	var r elasticResponse
+	if err := json.NewDecoder(resp.Body).Decode(&r); err != nil {
+		return nil, errors.Wrap(err, "error parsing the response body")
+	}
+	serp := &service.ResultPage{
+		Total:     uint64(r.Hits.Total.Value),
+		Documents: make([]service.ResultDoc, 0, len(r.Hits.Hits)),
+	}
+	for _, v := range r.Hits.Hits {
+		serp.Documents = append(serp.Documents, service.ResultDoc{
+			PostURL: v.Source.URL,
+			ID:      v.Source.ID,
+		})
+	}
+	return serp, nil
+}
+
+func (e *elastic) initSite(ctx context.Context, siteID string, eng engine.Interface) error {
+	resp, err := e.client.Indices.Exists([]string{siteID})
+	if err != nil {
+		return errors.Wrapf(err, "error getting index status")
+	}
+
+	if closeErr := resp.Body.Close(); closeErr != nil {
+		log.Printf("[ERROR] error to close response body %v", closeErr)
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		log.Printf("[INFO] site %q exists in index, skipping", siteID)
+		return nil
+	}
+	if resp.StatusCode != http.StatusNotFound {
+		return checkElasticResponseErr(resp)
+	}
+
+	resp, err = e.client.Indices.Create(
+		siteID,
+		e.client.Indices.Create.WithBody(e.buildCreateIndexSettings()),
+	)
+	if err != nil {
+		return errors.Wrapf(err, "error create index")
+	}
+
+	if closeErr := resp.Body.Close(); closeErr != nil {
+		log.Printf("[ERROR] error to close response body %v", closeErr)
+	}
+
+	if err = checkElasticResponseErr(resp); err != nil {
+		return errors.Wrapf(err, "error create index")
+	}
+
+	idxr := &siteIndexer{
+		parent: e,
+		siteID: siteID,
+	}
+	return indexSite(ctx, siteID, eng, idxr)
+}
+
+// Init elastic engine
+func (e *elastic) Init(ctx context.Context, eng engine.Interface) error {
+	errs := new(multierror.Error)
+
+	for siteID := range e.bulkIndexers {
+		err := e.initSite(ctx, siteID, eng)
+		errs = multierror.Append(err, errs)
+	}
+
+	err := errs.ErrorOrNil()
+	if err == nil {
+		e.ready = true
+	}
+
+	return err
+}
+
+// Delete comment from index
+func (e *elastic) Delete(siteID, commentID string) error {
+	if bi, has := e.bulkIndexers[siteID]; has {
+		item := esutil.BulkIndexerItem{
+			Action:     "delete",
+			DocumentID: commentID,
+			OnFailure: func(ctx context.Context, item esutil.BulkIndexerItem, res esutil.BulkIndexerResponseItem, err error) {
+				if err != nil {
+					log.Printf("[ERROR] failed to delete document: %s", err)
+					return
+				}
+				log.Printf("[ERROR]: %s: %s", res.Error.Type, res.Error.Reason)
+			},
+		}
+		return bi.Add(e.ctx, item)
+	}
+	return errors.Errorf("index for site %s does not found", siteID)
+}
+
+// Flush all unprocessed documents
+func (e *elastic) Flush(siteID string) error {
+	// TODO(@vdimir)
```

Comment:
Can we leave this part for now? Actually I don't think that elastic would be used in production, but implementing it not so clear. So, can supporting flush for elastic be omitted for now?

Reply:
Yes, but please explicitly state what code there should be and why it's not written so it would be relatively easy to discover and fix if somebody will need that.

Reply:
Ok, I've added comment and return error from this method